graphbolt/src/csc_sampling_graph.cc

```diff
@@ -472,5 +496,233 @@ torch::Tensor PickByEtype(
   return torch::cat(picked_neighbors, 0);
 }
 
+template <>
+torch::Tensor Pick<SamplerType::LABOR>(
+    int64_t offset, int64_t num_neighbors, int64_t fanout, bool replace,
+    const torch::TensorOptions& options,
+    const torch::optional<torch::Tensor>& probs_or_mask,
+    SamplerArgs<SamplerType::LABOR> args) {
+  if (fanout == 0) return torch::tensor({}, options);
+  if (probs_or_mask.has_value()) {
+    torch::Tensor picked_neighbors;
+    AT_DISPATCH_FLOATING_TYPES(
+        probs_or_mask.value().scalar_type(), "LaborPickFloatType", ([&] {
+          if (replace) {
+            picked_neighbors = LaborPick<true, true, scalar_t>(
+                offset, num_neighbors, fanout, options, probs_or_mask, args);
+          } else {
+            picked_neighbors = LaborPick<true, false, scalar_t>(
+                offset, num_neighbors, fanout, options, probs_or_mask, args);
+          }
+        }));
+    return picked_neighbors;
+  } else if (replace) {
+    return LaborPick<false, true>(
+        offset, num_neighbors, fanout, options, probs_or_mask, args);
+  } else {  // replace = false
+    return LaborPick<false, false>(
+        offset, num_neighbors, fanout, options, probs_or_mask, args);
+  }
+}
+
+template <typename T>
+inline T labor_uniform_random(int64_t random_seed, int64_t t) {
```

Comment:
Move this to `random.h` under graphbolt folder as this seems a general function.